core/string/ustring.cpp

```diff
@@ -3655,6 +3655,30 @@ bool String::is_absolute_path() const {
 	}
 }
 
+static _FORCE_INLINE_ bool _is_valid_identifier(int p_index, char32_t p_char) {
+	if (p_index == 0 && is_digit(p_char)) {
+		return false; // No start with number plz.
+	}
+	return is_ascii_identifier_char(p_char);
+}
+
+String String::validate_identifier() const {
+	String result;
+
+	int len = length();
+	const char32_t *str = &operator[](0);
+
+	for (int i = 0; i < len; i++) {
+		if (_is_valid_identifier(i, str[i])) {
+			result += str[i];
+		} else {
+			result += "_";
+		}
+	}
```

Comment:
Building String char by char is rather expensive. Maybe it doesn't matter much for where this method is used, but it could be optimized by modifying the char array directly and then building a String from it.